xde/src/xde.rs

```diff
@@ -1860,6 +1859,7 @@ unsafe extern "C" fn xde_mc_propinfo(
 fn new_port(
     name: String,
     cfg: &VpcCfg,
+    v2p: Arc<overlay::Virt2Phys>,
```

Comment:
This seems like the heart of the change around the `Resource` trait, IIUC. Previously, this data was passed to the `Port::process` method as the last argument. Now, the V2P mapping is still maintained by the driver's soft-state, and a pointer to that is passed to each port at the time it's created. Is that a fair, if rough, summary?

Does the `overlay::Virt2Phys` object do its own internal synchronization? It seems like it has to, for it to be shared in an `Arc<_>` instead of an `Arc<Mutex<_>>`.

Reply:
Yes. Previously we placed this in the `Resources` map which was just anymap underneath; and we threaded that thru the entire callstack down to the action. The V2P mapping is still owned by xde, and each port's encap action gets a pointer/Arc to this mapping.

And yes, `Virt2Phys` performs its own synchronization since it's in an Arc.